lib/kubernetes-deploy/kubernetes_resource.rb

```diff
@@ -112,18 +114,6 @@ def group_name
       type.downcase.pluralize
     end
 
-    def run_kubectl(*args)
-      raise FatalDeploymentError, "Namespace missing for namespaced command" if namespace.blank?
-      raise FatalDeploymentError, "Explicit context is required to run this command" if context.blank?
```

Comment:
Also, the KubernetesResource version only returned two of the three args, so you'll need to accommodate that here or change all the callers.